network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/integration/NetpSettingsStateCollector.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.integration
+
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
+import com.duckduckgo.mobile.android.vpn.state.VpnStateCollectorPlugin
+import com.duckduckgo.networkprotection.impl.NetPVpnFeature
+import com.duckduckgo.networkprotection.impl.settings.NetPSettingsLocalConfig
+import com.duckduckgo.networkprotection.store.NetPGeoswitchingRepository
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+import org.json.JSONObject
+
+@ContributesMultibinding(ActivityScope::class)
+class NetpSettingsStateCollector @Inject constructor(
+    private val vpnFeaturesRegistry: VpnFeaturesRegistry,
+    private val netPSettingsLocalConfig: NetPSettingsLocalConfig,
+    private val netPGeoswitchingRepository: NetPGeoswitchingRepository,
+) : VpnStateCollectorPlugin {
+    override suspend fun collectVpnRelatedState(appPackageId: String?): JSONObject {
+        return JSONObject().apply {
+            if (vpnFeaturesRegistry.isFeatureRunning(NetPVpnFeature.NETP_VPN)) {
+                put("customServerSelection", netPGeoswitchingRepository.getUserPreferredLocation().countryCode != null)
+                put("excludeLocalNetworksEnabled", netPSettingsLocalConfig.vpnExcludeLocalNetworkRoutes().isEnabled())
+            }
```

Comment:
Should we move both `customServerSelection` and `excludeLocalNetworksEnabled` inside the NetPStateCollector instead of creating a new JSON struct at root level?